### PR TITLE
Add definitions for okta-vue

### PR DESCRIPTION
**Pull request by mgdodge**
Please fill in this template.

- [ xx] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x ] Test the change in your own code. (Compile and run.)
- [ x ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ x ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ x ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ x ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ x ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by mgdodge** [comment]
@rbuckton You commented on another PR I just submitted today...just curious if you could explain to me what the error here means. Not sure I understand.

File: types/okta__okta-vue/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { NavigationGuard } from 'vue-router';
+
+export interface OktaVueOptions {
+    issuer: string;
+    client_id: string;
+    redirect_uri: string;
+    scope: string;
+}
+export function OktaVuePlugin(vm: VueConstructor, options: OktaVueOptions): void;
+export namespace OktaVuePlugin {
```
Comment (rbuckton) [inline types/okta__okta-vue/index.d.ts]: It looks like the plugin is actually an object with an `install` and a `handleCallback`: https://github.com/okta/okta-oidc-js/blob/master/packages/okta-vue/src/Auth.js#L93

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/okta__okta-vue/index.d.ts (hunk 1 above)
Comment (mgdodge) [reply types/okta__okta-vue/index.d.ts]: Making update.

**Comment by mgdodge** [comment]
Bump - @rbuckton still waiting for review following updates.

File: types/okta__okta-vue/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Type definitions for okta-vue 1.0
+// Project: https://github.com/okta/okta-oidc-js/tree/master/packages/okta-vue
+// Definitions by: Mike Dodge <https://github.com/innovation-team>
```
Comment (andy-ms) [inline types/okta__okta-vue/index.d.ts]: Looks like your username is `mgdodge` and not `innovation-team`?

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**